File: Cpp/Meituan/t4.py
n = int(input())
a = [int(x) for x in input().split()]
if n < 3:
    print(0)
else:
    ans = 0
    cnt = {}
    cnt[3 * a[n - 2] - a[n - 1]] = 1
    for i in range(n - 3, -1, -1):
        if a[i] in cnt:
            ans += cnt[a[i]]
        for j in range(i + 1, n):
            if 3 * a[i] - a[j] not in cnt:
                cnt[3 * a[i] - a[j]] = 1
            else:
                cnt[3 * a[i] - a[j]] += 1
    print(ans)

File: Cpp/Meituan/t4.cpp
#include <iostream>
#include <unordered_map>

using ll = long long;
using namespace std;
int a[4005];

int main(int argc, char const *argv[])
{
    int n;
    cin >> n;
    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }
    if (n < 3) {
        cout << 0 << endl;
        return 0;
    }
    unordered_map<int, ll> cnt;
    ll ans = 0;
    cnt[3 * a[n - 2] - a[n - 1]] = 1;
    for (int i = n - 3; i >= 0; --i) {
        ans += cnt[a[i]];
        for (int j = i + 1; j < n; ++j) {
            ++cnt[3 * a[i] - a[j]];
        }
    }
    cout << ans << endl;
    return 0;
}
